cql3/query_processor.cc

```diff
@@ -1179,4 +1179,39 @@ void query_processor::reset_cache() {
     _authorized_prepared_cache.reset();
 }
 
+future<> query_processor::truncate_table(sstring ks_name, sstring table_name, std::chrono::milliseconds timeout_in_ms) {
```

Comment:
This belongs in storage_proxy, no?

Reply:
I think the best place for this would be `truncate_statement::execute()`. But we need a reference to `storage_service` to obtain the ERM pointer. `query_processor` has this reference, but `storage_proxy` does not. Which is why I chose `query_processor` to put this in. I though about adding a public member `query_processor::get_ss()`, and then put the truncate operation in `truncate_statement::execute()`, but I decided against that because of the need for holding a gate in `query_processor::remote`.

I chose what seemed to be the least intrusive solution.

Reply:
The way to think about this is: is this specific to a front-end, or is it common? TRUNCATE is not specific to CQL (at least conceptually). Would we reimplement it for alternator if DynamoDB had an equivalent to TRUNCATE? no.

I'm sure storage_proxy methods can obtain an effective replication map ~pointer~lock.

Reply:
I have moved this to `storage_proxy::truncate_blocking`

Reply:
Closing conversation as this design was abandoned.